src/main/java/com/salesforce/dynamodbv2/mt/mappers/MtAmazonDynamoDbStreamsByTable.java

```diff
@@ -17,9 +23,31 @@
     }
 
     @Override
-    protected Function<Record, MtRecord> getRecordMapper(StreamArn arn) {
-        String[] tenantAndTableName = mtDynamoDb.getTenantAndTableName(arn.getTableName());
-        return record -> mapRecord(tenantAndTableName[0], tenantAndTableName[1], record);
+    protected MtGetRecordsResult getMtRecords(GetRecordsRequest request, StreamArn streamArn) {
+        final String[] tenantAndTableName = mtDynamoDb.getTenantAndTableName(streamArn.getTableName());
+        return getMtRecords(request, tenantAndTableName[0], tenantAndTableName[1]);
+    }
+
+    @Override
+    protected MtGetRecordsResult getMtRecords(GetRecordsRequest request, MtStreamArn mtStreamArn) {
+        return getMtRecords(request, mtStreamArn.getContext(), mtStreamArn.getTenantTableName());
+    }
```

Comment:
If it's called in tenant context, then we can use the parsed values from the arn directly.

Other than that, the methods do no differ for this strategy.